### PR TITLE
fix #17898(alternative to #18149 randomPathName called twice in a row can return the same string on windows)

### LINKED ISSUE #17898: `randomPathName` called twice in a row can return the same string on windows
this is the 2nd bug with std/tempfiles i've encountered while working on https://github.com/nim-lang/Nim/pull/17892 (i already sent a fix for the 1st one https://github.com/nim-lang/Nim/issues/17888 which affected it). This shows that dog-fooding is useful and we should do more of it (ie, promote using stdlib within nim repo).

### Example
```nim
# see https://github.com/nim-lang/Nim/pull/17892 for `genTempPath`, which
# is based on `randomPathName`
proc diffStrings*(a, b: string): tuple[output: string, same: bool] =
  template tmpFileImpl(prefix, str): auto =
    let path = genTempPath(prefix, "")
    writeFile(path, str)
    path
  let patha = tmpFileImpl("diffStrings_a_", a)
  let pathb = tmpFileImpl("diffStrings_b_", b)
  defer:
    removeFile(patha)
    removeFile(pathb)
  result = diffFiles(patha, pathb)
```

### Current Output
on windows, in CI, `echo diffStrings(lhs, rhs)` produced the following:
```
2021-04-29T20:00:01.1496001Z --- "a/C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\diffStrings_a_C5xLA9E2"
2021-04-29T20:00:01.1507701Z +++ "b/C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\diffStrings_b_C5xLA9E2"
```
(see also https://gist.github.com/timotheecour/515dde92cb29ce359a3f4b9546c247fd)

it looks like `randomPathName` produced the same string `C5xLA9E2`, and this is unlikely to be by accident.

### Expected Output
`randomPathName` called twice in a row should return different strings with very high probability.

### Additional Information
1.5.1 16405083485967a395b8d677833bc26040881b21

I'm not on windows, but windows users should be able to minimize this issue with a reproducible example based on this, eg by calling `randomPathName` in a loop and checking for nearest neighbor duplicates

/cc @xflywind can you reproduce this on your end?

### PR DESCRIPTION
**Pull request by ringabout**
close #17898

### CODE REVIEW

File: lib/std/private/gitutils.nim [hunk 1]
```diff
@@ -4,7 +4,8 @@ internal API for now, API subject to change
 
 # xxx move other git utilities here; candidate for stdlib.
 
-import std/[os, osproc, strutils, tempfiles]
+import std/[os, osproc, strutils, random]
+from std/tempfiles import genTempPath
```
Comment (ringabout) [inline lib/std/private/gitutils.nim:8]: `from std/tempfiles {.all.} import genTempPath` seems to cause booting issue. So I have to public `genTempPath`.

**Comment by timotheecour** [comment]
instead of requiring client code to pass in a Rand state, how about let's use atomics, as in oids

**Comment by Araq** [comment]
Passing the rand state is fine... :-/

**Comment by juancarlospaco** [comment]
I like the var Rand and the inline, I dont know why this API is not in-place, but Ok.

**Comment by ringabout** [comment]
Succeeded by https://github.com/nim-lang/Nim/pull/18729

**Pull request closed without merging**